articles/js-execution-under-the-hood/index.md

```diff
@@ -0,0 +1,167 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/js-execution-under-the-hood
+title: What Happens When JavaScript Runs My Code?
+description: This guide walks you through understanding what happens when the JavaScript engine runs your code.
+author: adrian-murage
+date: 
+topics: 
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/js-execution-under-the-hood/hero.jpg
+    alt: JS execution under the hood
+---
+
+### Introduction
+JavaScript is a single-threaded language, where only one command executes at a time. And, it has a Synchronous model of execution, where each line executes line by line, top to bottom.
+
+In this guide, you get to understand what happens as the JavaScript engine threads down your code.
+
+## Prerequisites
+
+Before you begin this guide you'll need the following:
+
+- A basic understanding of JavaScript Functions
+
+## Step 1 — What Happens When JavaScript Executes My Code?
+
+```javascript 
+1. const num = 3;
+2. function multiplyBy2 (inputNumber){
+3.     const result = inputNumber*2;
+4.     return result;
+5.     }
+6. const name = "Will"
+```
+
+In this step, you will walk through the above lines of code like the JavaScript engine would.
+
+Why?
+
+To understand what happens when you execute the code above.
```

Comment:
You may want to incorporate these into a paragraph, something like, "In this step, we will walk through the above lines of code like the Javascript engine would. This is so you will understand what happens when you execute the code above."